routers/web/repo/pull.go

```diff
@@ -738,24 +738,59 @@ func ViewPullCommits(ctx *context.Context) {
 	ctx.HTML(http.StatusOK, tplPullCommits)
 }
 
+func ViewPullCommitFiles(ctx *context.Context) {
+	ctx.Data["PageIsPullCommits"] = true
+	issue := checkPullInfo(ctx)
+	if ctx.Written() {
+		return
+	}
+	var prInfo *git.CompareInfo
+	if issue.PullRequest.HasMerged {
+		prInfo = PrepareMergedViewPullInfo(ctx, issue)
+	} else {
+		prInfo = PrepareViewPullInfo(ctx, issue)
+	}
+
+	if ctx.Written() {
```

Comment:
https://github.com/go-gitea/gitea/pull/26197/files#diff-9107a2d87c7913bc4fbdf735c0391d8e9ed5936ebac72a9bd23188616accc4a7R744-R746 
same check, is ctx is modified

Reply:
Yes. But it's still necessary because `PrepareViewPullInfo` is invoked.